sdks/python/apache_beam/ml/rag/types.py

```diff
@@ -0,0 +1,74 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""Core types for RAG pipelines.
+This module contains the core dataclasses used throughout the RAG pipeline
+implementation, including Chunk and Embedding types that define the data
+contracts between different stages of the pipeline.
+"""
+
+import uuid
+from dataclasses import dataclass
+from dataclasses import field
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Tuple
+
+
+@dataclass
+class Content:
+  """Container for embeddable content.
+    """
+  text: Optional[str] = None
+  image_data: Optional[bytes] = None
```

Comment:
I'm leaning towards not including this in v1 since we don't have anything that knows how to deal with it. I still think having the dataclass makes sense, and we can append fields as we go (we can call this out in the docstring)